lib/streams/batch-stream.js

```diff
@@ -0,0 +1,45 @@
+var Transform = require('../transform');
+var inherits = require('util').inherits;
+
+/**
+ * BatchStream batches the incoming data into an array of size `batchSize`
+ * NOTE: objectMode will always be true
+ *
+ * @class BatchStream
+ * @constructor
+ * @extends Transform
+ * @param {Integer} [batchSize=10] - Size of the batches
+ * @param {Object} [options] - Stream options object
+ */
+function BatchStream(batchSize, options) {
+	if(typeof batchSize === 'object') {
+		options = batchSize;
+		batchSize = null;
+	}
+	batchSize = parseInt(batchSize);
+	batchSize = isNaN(batchSize) ? 10 : batchSize;
+	options = !options ? {} : options;
+
+	options.objectMode = true;
+	Transform.call(this, options);
+	
+	this._batchSize = batchSize;
+	this._buffer = [];
```

Comment:
You may want to call this something else to avoid possible conflicts with the internals of the superclasses ...